nautobot/core/tests/test_api.py

```diff
@@ -838,3 +839,68 @@ def test_get(self):
         }
         for view_name, view_kwarg in view_kwargs.items():
             self.assertEqual(view_name, get_view_name(viewset(**view_kwarg)))
+
+
+@tag("api")
+class RenderJinjaViewTest(testing.TestCase):
+    """Test case for the RenderJinjaView API view."""
+
+    def setUp(self):
+        """
+        Create a token for API calls.
+        """
+        super().setUp()
+        self.client.logout()
+        self.token = Token.objects.create(user=self.user)
+        self.header = {"HTTP_AUTHORIZATION": f"Token {self.token.key}"}
+
+    def test_render_jinja_template(self):
+        """
+        Test rendering a valid Jinja template.
+        """
+        response = self.client.post(
+            reverse("core-api:render_jinja_template"),
+            {
+                "template_code": r"{{ foo }}",
+                "context": {"foo": "bar"},
+            },
+            format="json",
+            **self.header,
+        )
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertSequenceEqual(
+            list(response.data.keys()),
+            ["rendered_template", "rendered_template_lines", "template_code", "context"],
+        )
+        self.assertEqual(response.data["rendered_template"], "bar")
+        self.assertEqual(response.data["rendered_template_lines"], ["bar"])
```

Comment:
Might be good to also have:

- test for multi-line template (distinguishing behavior of `rendered_template` and `rendered_template_lines` a bit further)
- test for a template that includes usage of some of Nautobot's built-in template filters, since that's a desired piece of the feature.

Reply:
It looks like we're only exposing the netutils filters to the django template engine. We're probably going to want those exposed to the jinja engine for this feature?
https://github.com/nautobot/nautobot/blob/develop/nautobot/core/templatetags/netutils.py

Reply:
They *should* be exposed to both already. Are you seeing behavior that demonstrates otherwise?

Reply:
https://github.com/nautobot/nautobot/blob/develop/nautobot/core/apps/__init__.py#L886-L888

Reply:
Thanks! I didn't expect them to be in two different places but it makes sense now that I've looked at #3296